software/database/src/main/java/brooklyn/entity/database/mysql/MySqlClusterImpl.java

```diff
@@ -0,0 +1,445 @@
+                .configure(MySqlNode.CREATION_SCRIPT_CONTENTS, (String) null);
+    }
+
+    private EntitySpec<?> applyDefaults(EntitySpec<?> spec, Supplier<Integer> serverId, String configUrl, boolean resetCreationScript) {
+        boolean needsServerId = !isKeyConfigured(spec, MySqlNode.MYSQL_SERVER_ID);
+        boolean needsConfigUrl = !isKeyConfigured(spec, MySqlNode.TEMPLATE_CONFIGURATION_URL.getConfigKey());
+        boolean needsCreationScriptUrl = resetCreationScript && !isKeyConfigured(spec, MySqlNode.CREATION_SCRIPT_URL);
+        boolean needsCreationScriptContents = resetCreationScript && !isKeyConfigured(spec, MySqlNode.CREATION_SCRIPT_CONTENTS);
+        if (needsServerId || needsConfigUrl || needsCreationScriptUrl || needsCreationScriptContents) {
+            EntitySpec<?> clonedSpec = EntitySpec.create(spec);
+            if (needsServerId) {
+                clonedSpec.configure(MySqlNode.MYSQL_SERVER_ID, serverId.get());
+            }
+            if (needsConfigUrl) {
+                clonedSpec.configure(MySqlNode.TEMPLATE_CONFIGURATION_URL, configUrl);
+            }
+            if (needsCreationScriptUrl) {
+                clonedSpec.configure(MySqlNode.CREATION_SCRIPT_URL, (String) null);
+            }
+            if (needsCreationScriptContents) {
+                clonedSpec.configure(MySqlNode.CREATION_SCRIPT_URL, (String) null);
+            }
+            return clonedSpec;
+        } else {
+            return spec;
+        }
+    }
+
+    private boolean isKeyConfigured(EntitySpec<?> spec, ConfigKey<?> key) {
+        return spec.getConfig().containsKey(key) || spec.getFlags().containsKey(key.getName());
+    }
+
+    @Override
+    protected Entity createNode(Location loc, Map<?, ?> flags) {
+        Entity node = super.createNode(loc, flags);
+        if (!IS_MASTER.apply(node)) {
+            ServiceNotUpLogic.updateNotUpIndicator((EntityLocal)node, MySqlSlave.SLAVE_HEALTHY, "Replication not started");
+
+            FunctionFeed.builder()
+                .entity((EntityLocal)node)
+                .period(Duration.FIVE_SECONDS)
+                .poll(FunctionPollConfig.forSensor(MySqlSlave.SLAVE_HEALTHY)
+                        .callable(new SlaveStateCallable(node))
+                        .checkSuccess(StringPredicates.isNonBlank())
+                        .onSuccess(new SlaveStateParser(node))
+                        .setOnFailure(false)
+                        .description("Polls SHOW SLAVE STATUS"))
+                .build();
+
+            node.addEnricher(Enrichers.builder().updatingMap(Attributes.SERVICE_NOT_UP_INDICATORS)
+                    .from(MySqlSlave.SLAVE_HEALTHY)
+                    .computing(Functionals.ifNotEquals(true).value("Slave replication status is not healthy") )
+                    .build());
+        }
+        return node;
+    }
+
+    public class SlaveStateCallable implements Callable<String> {
```

Comment:
Can we make this inner-class static?

Will the `FunctionFeed` that uses this get persisted? I don't think so because it's not using `AbstractEntity`'s `addFeed(feed)`. Does that mean it will stop populating `SLAVE_HEALTHY` after a brooklyn reboot?